sycl/source/detail/scheduler/commands.cpp

```diff
@@ -729,17 +695,20 @@ Command *Command::processDepEvent(EventImplPtr DepEvent, const DepDesc &Dep,
 
   ContextImplPtr DepEventContext = DepEvent->getContextImpl();
   // If contexts don't match we'll connect them using host task
-  if (DepEventContext != WorkerContext && !WorkerContext->is_host()) {
+  if (DepEventContext == WorkerContext)
+    MPreparedDepsEvents.push_back(std::move(DepEvent));
+  else
+  {
     Scheduler::GraphBuilder &GB = Scheduler::getInstance().MGraphBuilder;
     ConnectionCmd = GB.connectDepEvent(this, DepEvent, Dep, ToCleanUp);
-  } else
-    MPreparedDepsEvents.push_back(std::move(DepEvent));
+  }
 
   return ConnectionCmd;
 }
 
 const ContextImplPtr &Command::getWorkerContext() const {
-  return MQueue->getContextImplPtr();
+  assert(MWorkerQueue && "MWorkerQueue must not be nullptr");
```

Comment:
I changed queue here, need to check usages. probably it was not a bug but intended replacement with mQueue

Reply:
fixed